lib/utilities/valid-component-name.js

```diff
@@ -0,0 +1,12 @@
+'use strict';
+var SilentError = require('../errors/silent');
+
+module.exports = function(entityName) {
```

Comment:
can you add an explicit test for this new extracted module?

basically for the two branches:

* test that it throws, with the correct error type and message
* test that it returns entityName